openverse_catalog/dags/providers/provider_api_scripts/science_museum.py

```diff
@@ -45,198 +27,202 @@
+
+    @staticmethod
+    def _get_dimensions(image_data: Dict) -> Tuple[int | None, int | None]:
+        """
+        Returns the height and width of the image from "image_data"."measurements"
+        with keys of "dimension", "units", "value".
+        """
+        size = {}
+        dimensions = image_data.get("measurements", {}).get("dimensions")
         if dimensions:
             for dim in dimensions:
-                height_width[dim.get("dimension")] = dim.get("value")
-    return height_width.get("height"), height_width.get("width")
-
-
-def _get_license_version(source):
-    license_version = None
-    if source:
-        legal = source.get("legal")
-        if legal:
-            rights = legal.get("rights")
-            if type(rights) == list:
-                license_version = rights[0].get("usage_terms")
-    return license_version
-
-
-def _get_metadata(obj_attr):
-    metadata = {}
-    identifier = obj_attr.get("identifier")
-    if type(identifier) == list:
-        metadata["accession number"] = identifier[0].get("value")
-    name = obj_attr.get("name")
-    if type(name) == list:
-        metadata["name"] = name[0].get("value")
-    category = obj_attr.get("categories")
-    if type(category) == list:
-        metadata["category"] = category[0].get("value")
-    creditline = obj_attr.get("legal")
-    if type(creditline) == dict:
-        metadata["creditline"] = creditline.get("credit_line")
-    description = obj_attr.get("description")
-    if type(description) == list:
-        metadata["description"] = description[0].get("value")
-    return metadata
+                size[dim.get("dimension")] = (
+                    dim.get("value") if dim.get("units") == "pixels" else None
+                )
+        return size.get("height"), size.get("width")
+
+    @staticmethod
+    def _get_image_info(
+        processed: Dict,
+    ) -> Tuple[Optional[str], Optional[int], Optional[int], Optional[str]]:
+        height, width, filetype = None, None, None
+        image_data = processed.get("large")
+        if image_data is None:
+            image_data = processed.get("medium", {})
+
+        image_url = ScienceMuseumDataIngester.check_url(image_data.get("location"))
+        if image_url:
+            filetype = image_data.get("format")
+            height, width = ScienceMuseumDataIngester._get_dimensions(image_data)
+        return image_url, height, width, filetype
+
+    @staticmethod
+    def _get_first_list_value(key: str, attributes: Dict) -> str | None:
+        val = attributes.get(key)
+        if isinstance(val, list):
+            return val[0].get("value")
+        return None
+
+    @staticmethod
+    def _get_metadata(attributes):
+        metadata = {}
+        for item in [
+            ("identifier", "accession number"),
+            ("name", "name"),
+            ("categories", "category"),
+            ("description", "description"),
+        ]:
+            attr_key, metadata_key = item
```

Comment:
This could just be:
```suggestion
        for attr_key, metadata_key in [
            ("identifier", "accession number"),
            ("name", "name"),
            ("categories", "category"),
            ("description", "description"),
        ]:
```